catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/RawUdpDataToMTSPacketDecoder.java

```diff
@@ -143,37 +147,41 @@ protected void decode(ChannelHandlerContext ctx, DatagramPacket msg, List<Object
         notNull(msg, "msg must be non-null");
         notNull(outputList, "outputList must be non-null");
 
-        checkSecuritySubject(msg);
+        LOCK.lock();
```

Comment:
Any that modifies the buffer, which includes reads and writes (reads modify the buffer state), needs to be protected. The only line that doesn't have to be protected is the checkSecuritySubject. I could change that one line of you want.